jbpm-case-mgmt/jbpm-case-mgmt-impl/src/test/java/org/jbpm/casemgmt/impl/SubCaseServiceImplTest.java

```diff
@@ -485,4 +486,52 @@ public void testStartCaseWithSubCaseAbortProcessInstanceOfSubCase() {
         }
     }
 
+    @Test
+    public void testCaseWithSubCaseQuery() {
+
+        Map<String, OrganizationalEntity> roleAssignments = new HashMap<>();
+        roleAssignments.put("owner", new UserImpl("john"));
+        roleAssignments.put("manager", new UserImpl("mary"));
+
+        Map<String, Object> data = new HashMap<>();
+        data.put("name", "John Doe");
+        CaseFileInstance caseFile = caseService.newCaseFileInstance(deploymentUnit.getIdentifier(), BASIC_SUB_CASE_P_ID, data, roleAssignments);
+
+        String caseId = caseService.startCase(deploymentUnit.getIdentifier(), BASIC_SUB_CASE_P_ID, caseFile);
+        assertNotNull(caseId);
+        assertEquals(SUB_CASE_ID, caseId);
+        try {
+            CaseInstance cInstance = caseService.getCaseInstance(caseId);
+            assertNotNull(cInstance);
+            assertEquals(deploymentUnit.getIdentifier(), cInstance.getDeploymentId());
+
+            caseService.triggerAdHocFragment(caseId, "Sub Case", null);
+
+            Collection<CaseInstance> caseInstances = caseRuntimeDataService.getCaseInstances(new QueryContext());
+            assertNotNull(caseInstances);
+            assertEquals(2, caseInstances.size());
+
+            // we check the parent
+            List<CaseStatus> allStatus = Arrays.asList(CaseStatus.values());
+            Collection<CaseInstance> subcases = caseRuntimeDataService.getSubCaseInstancesByParentCaseId(SUB_CASE_ID, allStatus, new QueryContext());
+            assertNotNull(subcases);
+            assertEquals(1, subcases.size());
+            List<String> childId = subcases.stream().map(CaseInstance::getCaseId).collect(Collectors.toList());
+            assertEquals(1, childId.size()); // only one child
+            assertTrue(childId.contains(HR_CASE_ID));
+            assertEquals(SUB_CASE_ID, subcases.iterator().next().getParentCaseId());
```

Comment:
SonarCloud still complaining about coverage under 80% :-( 
Include toString in some test case.

```suggestion
            CaseInstance sci = subcases.iterator().next();
            assertEquals(SUB_CASE_ID, sci.getParentCaseId());
            assertTrue(sci.toString().contains("parentCaseId="+SUB_CASE_ID));
```